src/WebJobs.Script.Abstractions/Description/FunctionMetadata.cs

```diff
@@ -7,7 +7,7 @@
 using System.IO;
 using System.Linq;
 
-namespace Microsoft.Azure.WebJobs.Script.Description
+namespace Microsoft.Azure.WebJobs.Script.Abstractions.Description
 {
     public class FunctionMetadata
```

Comment:
We should be able to remove IsProxy from this type, and use an internal subclass for proxies. Then everywhere we check IsProxy currently, we just do a type check. With this being a public type 3rd parties are coding against now, we don't want them to have to think about this. Same goes for IsDirect and IsHttpInAndOutFunction.